contrib/babelfishpg_tsql/src/backend_parser/gram-tsql-epilogue.y.c

```diff
@@ -2068,6 +2077,253 @@ tsql_pivot_select_transformation(List *target_list, List *from_clause, List *piv
+		List *prev_unpivot = (List *)table_ref;
+		if (list_length(prev_unpivot) == 7 &&
+			IsA(linitial(prev_unpivot), String) &&
+			strcmp(strVal(linitial(prev_unpivot)), "UNPIVOT") == 0)
+		{
+			/* Use the alias from previous UNPIVOT */
+			return strVal(list_nth(prev_unpivot, 1));
+		}
+
+	}
+	return NULL;
+}
+
+/*
+ * Transform TSQL UNPIVOT operation into equivalent CROSS JOIN LATERAL structure.
+ * Builds a JoinExpr node representing the transformation and returns additional
+ * context for analyzer stage processing.
+ *
+ * Input: components = list_make3(table_ref, unpivot_clause, alias)
+ * Returns: List of ["UNPIVOT", unpivot_alias, dimension_col, measure_col, 
+ *				   source_alias, source_cols, transformed_node]
+ *
+ * The function:
+ * - Extracts measure column, dimension column and source columns from unpivot clause
+ * - Generates/validates source table alias
+ * - Creates VALUES list for LATERAL subquery from source columns
+ * - Builds complete JOIN structure with proper aliases
+ */
+static Node *
+tsql_unpivot_transformation(List *components)
+{
+	Node *table_ref;
+	List *unpivot_info;
+	Alias *alias;
+	Node *measure_col;
+	Node *dim_col;
+	List *source_cols;
+	JoinExpr *n;
+	RangeSubselect *rarg;
+	SelectStmt *values_subquery;
+	List *values_list = NIL;
+	ListCell *lc;
+	char *source_alias;
+	List *result_info;
+	char *measure_colname;
+	char *dim_colname;
+	char *inner_alias_name;
+	
+	/* Extract components */
+	table_ref = (Node *)linitial(components);
+	unpivot_info = (List *)lsecond(components);
+	alias = (Alias *)lthird(components);
+	
+	/* Extract unpivot components */
+	measure_col = (Node *)linitial(unpivot_info);
+	dim_col = (Node *)lsecond(unpivot_info);
+	source_cols = (List *)lthird(unpivot_info);
+	
+	measure_colname = strVal(llast(((ColumnRef *)measure_col)->fields));
+	dim_colname = strVal(llast(((ColumnRef *)dim_col)->fields));
+	
+	/* Create basic nodes for equivalent CROSS JOIN LATERAL query tree */
+	n = makeNode(JoinExpr);
+	rarg = makeNode(RangeSubselect);
+	values_subquery = makeNode(SelectStmt);
+	
+	n->jointype = JOIN_INNER;
+	n->isNatural = false;
+	//source_alias = "invalid_default_alias"; /* Will be overwritten or error thrown */ 
+	/* Set up left side with alias if not present */
+	source_alias = get_unpivot_source_alias(table_ref);
+
+	if(source_alias == NULL){
+		ereport(ERROR,
+			(errcode(ERRCODE_SYNTAX_ERROR),
+			errmsg("Invalid source structure for UNPIVOT operation")));
+	}
+	n->larg = table_ref;
+	
+	/* Build VALUES list from unpivot source columns */
+	foreach(lc, source_cols)
+	{
+		// TODO: needs to be llast if aliases are used in values list, ex c.q1
+		String *col_name = (String *)lfirst(lc);
+		List *value_pair;
+		ColumnRef *col_ref;
+		
+		/* Create ColumnRef with source table's alias */
+		col_ref = makeNode(ColumnRef);
+		col_ref->fields = list_make2(
+			makeString(source_alias), /* source table alias */
+			makeString(strVal(col_name)) /* column name */
```

Comment:
col_name is already a String, why have to do this ? Is it for copying a string ?